core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BpmBillingProjectCreator.scala

```diff
@@ -0,0 +1,57 @@
+package org.broadinstitute.dsde.rawls.billing
+
+import akka.http.scaladsl.model.StatusCodes
+import org.broadinstitute.dsde.rawls.RawlsExceptionWithErrorReport
+import org.broadinstitute.dsde.rawls.model.{CreateRawlsV2BillingProjectFullRequest, ErrorReport, UserInfo}
+
+import scala.concurrent.{ExecutionContext, Future, blocking}
+
+
+/**
+ * This class knows how to validate Rawls billing project requests and instantiate linked billing profiles in the
+ * billing profile manager service.
+ */
+class BpmBillingProjectCreator(billingRepository: BillingRepository,
+                               billingProfileManagerDAO: BillingProfileManagerDAO)
+                              (implicit val executionContext: ExecutionContext) extends BillingProjectCreator {
+
+  override def validateBillingProjectCreationRequest(createProjectRequest: CreateRawlsV2BillingProjectFullRequest,
+                                                     userInfo: UserInfo): Future[Unit] = {
+    val azureManagedAppCoordinates = createProjectRequest.billingInfo match {
+      case Left(_) => throw new NotImplementedError("Google billing accounts not supported in billing profiles")
+      case Right(coords) => coords
+    }
+
+    for {
```

Comment:
Do we need the for/yield here?

Reply:
As mentioned in a previous comment, I think this check is redundant given what we're doing in BPM, I'll remove this. See my subsequent comment re: sequencing futures in a for comprehension.